From ae9072bbd4abfe3c0f7f419a5ab3e087dc4584b4 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Fri, 13 Jan 2012 20:29:37 -0500
Subject: [PATCH] build-sys: Fix some srcdir != builddir issues

* Look in the build directory for generated headers.
* Look in source directory for source files
* Also while we have the patient open, make some generation rules
  atomic.
---
 src/Makefile.am |   17 +++++++++--------
 1 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 869decd..322861a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -9,6 +9,7 @@ SUBDIRS = \
 INCLUDES =							\
 	-I.							\
 	-I$(srcdir)						\
+	-I$(builddir)						\
 	$(CONSOLE_KIT_CFLAGS)					\
 	$(POLKIT_CFLAGS)					\
 	$(DISABLE_DEPRECATED_CFLAGS)				\
@@ -83,19 +84,19 @@ dbusif_DATA = 					\
 	org.freedesktop.ConsoleKit.Session.xml	\
 	$(NULL)
 
-ck-manager-glue.h: org.freedesktop.ConsoleKit.Manager.xml Makefile.am
-	dbus-binding-tool --prefix=ck_manager --mode=glib-server --output=ck-manager-glue.h org.freedesktop.ConsoleKit.Manager.xml
-ck-seat-glue.h: org.freedesktop.ConsoleKit.Seat.xml Makefile.am
-	dbus-binding-tool --prefix=ck_seat --mode=glib-server --output=ck-seat-glue.h org.freedesktop.ConsoleKit.Seat.xml
-ck-session-glue.h: org.freedesktop.ConsoleKit.Session.xml Makefile.am
-	dbus-binding-tool --prefix=ck_session --mode=glib-server --output=ck-session-glue.h org.freedesktop.ConsoleKit.Session.xml
+ck-manager-glue.h: $(srcdir)/org.freedesktop.ConsoleKit.Manager.xml Makefile.am
+	dbus-binding-tool --prefix=ck_manager --mode=glib-server --output=ck-manager-glue.h $(srcdir)/org.freedesktop.ConsoleKit.Manager.xml
+ck-seat-glue.h: $(srcdir)/org.freedesktop.ConsoleKit.Seat.xml Makefile.am
+	dbus-binding-tool --prefix=ck_seat --mode=glib-server --output=ck-seat-glue.h $(srcdir)/org.freedesktop.ConsoleKit.Seat.xml
+ck-session-glue.h: $(srcdir)/org.freedesktop.ConsoleKit.Session.xml Makefile.am
+	dbus-binding-tool --prefix=ck_session --mode=glib-server --output=ck-session-glue.h $(srcdir)/org.freedesktop.ConsoleKit.Session.xml
 
 ck-marshal.c: ck-marshal.list
 	echo "#include \"ck-marshal.h\"" > $@ && \
-	@GLIB_GENMARSHAL@ $< --prefix=ck_marshal --body >> $@
+	@GLIB_GENMARSHAL@ $< --prefix=ck_marshal --body > $@.tmp && mv $@.tmp $@
 
 ck-marshal.h: ck-marshal.list
-	@GLIB_GENMARSHAL@ $< --prefix=ck_marshal --header > $@
+	@GLIB_GENMARSHAL@ $< --prefix=ck_marshal --header > $@.tmp && mv $@.tmp $@
 
 console_kit_daemon_SOURCES =	\
 	main.c			\
-- 
1.7.6.4

